app/main/views/suppliers.py

```diff
@@ -23,10 +23,28 @@ def list_suppliers():
 
     prefix = request.args.get('prefix', '')
 
-    suppliers = Supplier.query.order_by(Supplier.name)
+    framework = request.args.get('framework')
+
+    if framework:
+        is_valid_string_or_400(framework)
```

Comment:
Is the idea that we call this method for all URL query parameters as a basic check before proceeding?  Or is framework a special case for some reason?  If it's only our apps (our API client) calling the API do we really need this check here? Shouldn't we be checking user-supplied params at the frontend routes rather than in the API?